lib/models/blueprint.js

```diff
@@ -429,13 +441,72 @@ let Blueprint = CoreObject.extend({
     // gather fileInfos to be processed
     let fileInfos = await process.call(this, intoDir, locals);
 
+    const transformed = await Promise.all(
+      fileInfos.map((info) => {
+        if (this.shouldConvertToJS(options, info)) {
+          return this.convertToJS(info);
+        }
+
+        return info;
+      })
+    );
+
     // commit changes for each FileInfo (with prompting as needed)
-    await Promise.all(fileInfos.map((fi) => this._commit(fi)));
+    await Promise.all(transformed.map((info) => this._commit(info)));
 
     // run afterInstall/afterUninstall userland hooks
     await afterHook.call(this, options);
   },
 
+  /**
+    @private
+    @method shouldConvertToJS
+    @param {Object} options
+    @param {FileInfo} fileInfo
+    @return {Boolean}
+  */
+  shouldConvertToJS(options, fileInfo) {
+    // If this isn't turned on, it doesn't matter what else was passed, we're not touching it.
+    if (!this.shouldTransformTypeScript) {
+      return false;
+    }
+
+    // If the user wants TypeScript output but there is no TypeScript blueprint available, we want
+    // to warn them that they're not going to get what they're expecting while still at least giving
+    // them the JS output
+    if (options.typescript && !isTypeScriptFile(fileInfo.inputPath)) {
+      this.ui.writeLine(
+        chalk.yellow(
+          "You passed the '--typescript' flag but there is no TypeScript blueprint available. " +
+            'A JavaScript blueprint will be generated instead.'
+        )
+      );
+      return false;
+    }
+
+    // If it's a TypeScript file, we transform it *unless* the user has asked for TypeScript either
+    // globally (by setting `isTypeScriptProject` to true) or locally (by passing the `--typescript`
+    // flag when they invoked the generator).
+    return !(options.isTypeScriptProject || options.typescript) && isTypeScriptFile(fileInfo.inputPath);
```

Comment:
Let's beef this comment up slightly to indicate where we expect this to come from, and why it **isn't** just using `{ "typescript": true }`.

Also, maybe split into a variable declaration so we can explain the flag independently of the logic explanation.

Reply:
We should also consider adding `isTypeScriptProject` to the default `.ember-cli` file (with the default value being `false`)...